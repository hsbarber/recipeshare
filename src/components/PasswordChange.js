import React, { Component } from 'react';

import { auth } from '../firebase';

const passChangeState = {
  passwordOne: '',
  passwordTwo: '',
  invalid: true,
  error: null,
};

class PasswordChange extends Component {
  constructor(props) {
    super(props);

    this.state = { ...passChangeState };
    this.onChange = this.onChange.bind(this);
  }

  onChange = event => {
    const { passwordOne, passwordTwo } = this.state;
    this.setState({ [event.target.name]: event.target.value });
    if (passwordOne !== passwordTwo || passwordOne === '') {
      this.setState({ invalid: false });
    }
  };

  onSubmit = event => {
    const { passwordOne } = this.state;

    auth
      .doPasswordUpdate(passwordOne)
      .then(() => {
        this.setState({ ...passChangeState });
      })
      .catch(error => {
        this.setState({ error });
      });

    event.preventDefault();
  };

  render() {
    const { passwordOne, passwordTwo, invalid, error } = this.state;

    return (
      <form onSubmit={this.onSubmit}>
        <input
          value={passwordOne}
          onChange={event => this.onChange(event)}
          type="password"
          name="passwordOne"
          placeholder="New Password"
        />
        <input
          value={passwordTwo}
          onChange={event => this.onChange(event)}
          type="password"
          name="passwordTwo"
          placeholder="Confirm New Password"
        />
        <button disabled={invalid} type="submit">
          Reset My Password
        </button>

        {error && <p className="passwordError">{error.message}</p>}
      </form>
    );
  }
}

export default PasswordChange;
