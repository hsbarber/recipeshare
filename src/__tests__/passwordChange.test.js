import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PasswordChange from '../components/PasswordChange';
import '../enzyme';

describe('Password Change', () => {
  it('shallow renders without crashing', () => {
    shallow(<PasswordChange />);
  });
  it('component matches snapshot', () => {
    const wrapper = shallow(<PasswordChange />);
    expect(wrapper).toMatchSnapshot();
  });
  it('onchange is called when input is filled', () => {
    const wrapper = shallow(<PasswordChange />);
    const onChange = jest.spyOn(wrapper.instance(), 'onChange');
    const firstInput = wrapper.find('form input').first();
    firstInput.simulate('change', { target: { value: 'password1' } });
    wrapper.update();
    expect(onChange).toHaveBeenCalled();

    // expect(wrapper.state().passwordOne).toEqual('password1');
  });
  it('Change invalid state', () => {
    const wrapper = mount(<PasswordChange />);
    const button = wrapper.find('form button');
    expect(button.props().disabled).toBe(true);
    wrapper.setState({ invalid: false });
    expect(wrapper.state().invalid).toBe(false);
  });
  it('onSubmit called on form submit', () => {
    const currentUser = 'haganb3@gmail.com';
    const wrapper = mount(<PasswordChange auth={currentUser} />);
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };

    // const inputOne = wrapper.find('input').at(0);
    wrapper.setState({ passwordOne: 'passssword' });
    // inputOne.props().onChange({
    //   target: {
    //     value: '1234567900',
    //   },
    // });
    // const inputTwo = wrapper.find('input').at(1);
    wrapper.setState({ passwordTwo: 'passssword' });
    // inputTwo.props().onChange({
    //   target: {
    //     value: '1234567900',
    //   },
    // });
    const onSubmit = jest.spyOn(wrapper.instance(), 'onSubmit');
    //    const updatePassword = jest.spyOn(wrapper.instance(), 'updatePassword');
    const button = wrapper.find('form button');
    // wrapper.setState({ invalid: false });
    button.simulate('submit', fakeEvent);
    // expect(wrapper.find('.passwordError').length).toBe(1);
    // expect(onSubmit).toHaveBeenCalled();
    console.log(wrapper.state());
    console.log(wrapper.debug());
  });
});
