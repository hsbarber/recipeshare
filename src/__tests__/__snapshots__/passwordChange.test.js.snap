// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Change component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordChange />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="passwordOne"
          onChange={[Function]}
          placeholder="New Password"
          type="password"
          value=""
        />,
        <input
          name="passwordTwo"
          onChange={[Function]}
          placeholder="Confirm New Password"
          type="password"
          value=""
        />,
        <button
          disabled={true}
          type="submit"
        >
          Reset My Password
        </button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "passwordOne",
          "onChange": [Function],
          "placeholder": "New Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "passwordTwo",
          "onChange": [Function],
          "placeholder": "Confirm New Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset My Password",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Reset My Password",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="passwordOne"
            onChange={[Function]}
            placeholder="New Password"
            type="password"
            value=""
          />,
          <input
            name="passwordTwo"
            onChange={[Function]}
            placeholder="Confirm New Password"
            type="password"
            value=""
          />,
          <button
            disabled={true}
            type="submit"
          >
            Reset My Password
          </button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "passwordOne",
            "onChange": [Function],
            "placeholder": "New Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "passwordTwo",
            "onChange": [Function],
            "placeholder": "Confirm New Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset My Password",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Reset My Password",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
